src/main/java/de/tum/in/www1/artemis/web/rest/admin/AdminBuildJobQueueResource.java

```diff
@@ -77,11 +77,30 @@ public ResponseEntity<List<LocalCIBuildJobQueueItem>> getRunningBuildJobs() {
      */
     @GetMapping("build-agents")
     @EnforceAdmin
-    public ResponseEntity<List<LocalCIBuildAgentInformation>> getBuildAgentInformation() {
+    public ResponseEntity<List<LocalCIBuildAgentInformation>> getBuildAgentSummary() {
         log.debug("REST request to get information on available build agents");
-        List<LocalCIBuildAgentInformation> buildAgentInfo = localCIBuildJobQueueService.getBuildAgentInformation();
+        // remove the recentBuildJobs from the response
+        List<LocalCIBuildAgentInformation> buildAgentSummary = localCIBuildJobQueueService.getBuildAgentInformation().stream()
+                .map(agent -> new LocalCIBuildAgentInformation(agent.name(), agent.maxNumberOfConcurrentBuildJobs(), agent.numberOfCurrentBuildJobs(), agent.runningBuildJobs(),
+                        agent.status(), null))
+                .toList();
```

Comment:
That duplicates how the summary gets created (see LocalCIQueueWebsocketService). Can you please move this to a common service and call it from this resource and the websocket service?